### ISSUE TITLE
_cdf_distance bug?

### ISSUE THREAD

**Issue by jimhalverson** [scipy/scipy#10187]
In _cdf_distance in stats.py there is a block of the form

```
if u_weights is None:
        u_cdf = u_cdf_indices / u_values.size
```

In my experience with this code, u_cdf_indices is full of ints (as one would expect), so e.g. if 'u_cdf_indices = [0,1,2,3,4]' and 'u_values.size=4' this gives 'u_cdf = [0,0,0,0,1]' instead of 'u_cdf=[0,.2,.4,.6.,.8,1]'. 

Is the latter behavior desired? If so, then this seems like a bug, and a simple conversion to float will fix it.

**Comment by WarrenWeckesser** [comment]
@jimhalverson, thanks for keeping an eye on the code!  In this case, it is not a problem, because we have this line at the beginning of the file:

    from __future__ import division, print_function, absolute_import

By importing `division` from the future, we ensure that division of integers returns floating point values.

**Comment by jimhalverson** [comment]
Dear Warren,

Thanks very much for explaining it, I'm glad it isn't an issue!

Best
Jim

On Thu, May 16, 2019 at 2:03 PM Warren Weckesser <notifications@github.com<mailto:notifications@github.com>> wrote:

Closed #10187<https://nam05.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fscipy%2Fscipy%2Fissues%2F10187&data=02%7C01%7Cj.halverson%40northeastern.edu%7C2e85b2f662f94b30988108d6da28c947%7Ca8eec281aaa34daeac9b9a398b9215e7%7C0%7C0%7C636936266060489806&sdata=q9Dx%2B8tWtxeYy3y91K5q%2FzYaM4FP%2BzorlU5V9her81I%3D&reserved=0>.
